zerver/models.py

```diff
@@ -1589,9 +1589,20 @@ class UserBaseSettings(models.Model):
     enable_digest_emails = models.BooleanField(default=True)
     enable_login_emails = models.BooleanField(default=True)
     enable_marketing_emails = models.BooleanField(default=True)
-    realm_name_in_notifications = models.BooleanField(default=False)
     presence_enabled = models.BooleanField(default=True)
 
+    REALM_NAME_IN_NOTIFICATIONS_AUTOMATIC = 1
+    REALM_NAME_IN_NOTIFICATIONS_YES = 2
+    REALM_NAME_IN_NOTIFICATIONS_NO = 3
+    REALM_NAME_IN_NOTIFICATIONS_CHOICES = [
+        REALM_NAME_IN_NOTIFICATIONS_AUTOMATIC,
+        REALM_NAME_IN_NOTIFICATIONS_YES,
+        REALM_NAME_IN_NOTIFICATIONS_NO,
+    ]
+    realm_name_in_notifications = models.PositiveSmallIntegerField(
+        default=REALM_NAME_IN_NOTIFICATIONS_AUTOMATIC
+    )
```

Comment:
I think we should call the new field `realm_name_in_notifications_policy`; we generally use the `_policy` suffix in names for enum fields, to make it readable that they're not a boolean field. Similarly we should probably do names like `REALM_NAME_IN_NOTIFICATIONS_POLICY_AUTOMATIC`, `REALM_NAME_IN_NOTIFICATIONS_POLICY_NEVER`, `REALM_NAME_IN_NOTIFICATIONS_POLICY_ALWAYS`; I think that's more consistent with how we name other policy values.

One benefit of having a new name for this is that the database migration sequence because simpler -- you get to skip the `_new` business, and also we can more simply explain the API change in the API changelog -- previously there was a boolean thing with a different name.

We also need an API changelog entry and Changes notes in `zulip.yaml` for this API change.

See https://zulip.readthedocs.io/en/latest/documentation/api.html and/or browser a few commits here for details: `git log --follow api_docs/changelog.md`

Reply:
> we generally use the _policy suffix in names for enum fields, to make it readable that they're not a boolean field.

I wasn't aware of the pattern, but it makes sense -- **Updated the PR.**

Also, at some places in the codebase, we are using enum fields but they don't have `_policy` suffix.
For e.g. [demote_inactive_streams](https://github.com/zulip/zulip/blob/main/zerver/models.py#L1530) , [email_address_visibility](https://github.com/zulip/zulip/blob/main/zerver/models.py#L1612)

As stated, it will make them readable that they're not a boolean field.

We can take this as a follow-up PR -- please let me know.

> We also need an API changelog entry and Changes notes in zulip.yaml for this API change.

**Done**

Reply:
I realized while reading this against that the name should probably be `realm_name_in_email_notifications_policy` -- the `_notifications_` was ambiguous about what sort of notifications. Worth fixing now since we're already doing a migration to rename it.

Reply:
[CZO discussion](https://chat.zulip.org/#narrow/stream/378-api-design/topic/automate.20-.20include.20realm.20name.20in.20missedmessage.20email.20subject/near/1518719)